com.unity.shadergraph/Editor/ShaderFoundry/Utilities/ShaderGenerator.cs

```diff
@@ -29,15 +25,11 @@ void CollectUniqueProperties(Block block)
                 {
                     foreach (var prop in properties)
                     {
-                        var decl = prop.Attributes.GetDeclaration();
-                        if (decl == UnityEditor.ShaderGraph.Internal.HLSLDeclaration.DoNotDeclare)
+                        var propertyAttribute = PropertyAttribute.FindFirst(prop.Attributes);
+                        if (propertyAttribute != null && !propertyAttribute.Exposed)
```

Comment:
Do we need to add null property attributes to shaderProperties?

Reply:
All BlockVariables here should have a valid `[Property]` attribute (filtered via `block.Properties();`) but I put the check to be safe.